Tests failed. One node failed with this:
```
FAIL: test_get_choices_for_values (corehq.apps.userreports.tests.test_choice_provider.UserChoiceProviderTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_choice_provider.py", line 245, in test_get_choices_for_values
    ['unknown-user'] + [user._id for user in self.users])
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_choice_provider.py", line 102, in _test_get_choices_for_values
    self.static_choice_provider.get_choices_for_values(values),
AssertionError: Items in the first set but not the second:
Choice(value=None, display='')
Items in the second set but not the first:
Choice(value=None, display=u'dennis')
Choice(value=None, display=u'albert')
Choice(value=None, display=u'bernice')
Choice(value=None, display=u'elizabeth')
Choice(value=None, display=u'candice@example.com'
```

and two others failures:
```
ERROR: test suite for <class 'corehq.apps.reports.tests.test_project_health.ProjectHealthDashboardTest'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/vendor/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_project_health.py", line 179, in setUpClass
    cls.class_setup()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_project_health.py", line 33, in class_setup
    cls._setup_malt_tables()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_project_health.py", line 70, in _setup_malt_tables
    app_id=MISSING_APP_ID,
  File "/vendor/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 372, in create
    obj.save(force_insert=True, using=self.db)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 589, in save
    force_update=force_update, update_fields=update_fields)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 617, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 698, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 731, in _do_insert
    using=using, raw=raw)
  File "/vendor/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 921, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/vendor/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 921, in execute_sql
    cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
IntegrityError: null value in column "user_id" violates not-null constraint
DETAIL:  Failing row contains (1, 2016-10-01, null, testuser, , test_domain, 16, _MISSING_APP_ID, f, null, null, CommCareUser, 15, null, 3).
```

```
ERROR: test suite for <class 'corehq.apps.reports.tests.test_project_health.MonthlyPerformanceSummaryTests'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/vendor/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_project_health.py", line 100, in setUpClass
    cls.class_setup()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_project_health.py", line 33, in class_setup
    cls._setup_malt_tables()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_project_health.py", line 70, in _setup_malt_tables
    app_id=MISSING_APP_ID,
  File "/vendor/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 372, in create
    obj.save(force_insert=True, using=self.db)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 589, in save
    force_update=force_update, update_fields=update_fields)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 617, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 698, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 731, in _do_insert
    using=using, raw=raw)
  File "/vendor/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 921, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/vendor/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 921, in execute_sql
    cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
IntegrityError: null value in column "user_id" violates not-null constraint
DETAIL:  Failing row contains (1, 2016-10-01, null, testuser, , test_domain, 16, _MISSING_APP_ID, f, null, null, CommCareUser, 15, null, 3).
```

kicked all three just in case they're transient issues.

--- a/corehq/apps/es/fake/es_query_fake.py
+++ b/corehq/apps/es/fake/es_query_fake.py
@@ -46,7 +46,12 @@ class ESQueryFake(object):
     def save_doc(cls, doc):
         if '_id' not in doc:
             doc['_id'] = 'silly-fake-id-{}'.format(uuid.uuid4().hex[:8])
+        doc = cls.transform_doc(doc)
         cls._get_all_docs().append(doc)
+
+    @staticmethod
+    def transform_doc(doc):
+        return doc
 
     @classmethod
     def reset_docs(cls):

--- a/corehq/apps/es/fake/users_fake.py
+++ b/corehq/apps/es/fake/users_fake.py
@@ -16,8 +16,7 @@ class UserESFake(HQESQueryFake):
     def mobile_users(self):
         return self.term("doc_type", "CommCareUser")
 
-    @classmethod
-    def save_doc(cls, doc):
+    @staticmethod
+    def transform_doc(doc):
         doc['username.exact'] = doc.get('username', '')
-        doc = transform_user_for_elasticsearch(doc)
-        return super(UserESFake, cls).save_doc(doc)
+        return transform_user_for_elasticsearch(doc)

--- a/corehq/apps/reports/tests/test_project_health.py
+++ b/corehq/apps/reports/tests/test_project_health.py
@@ -40,8 +40,7 @@ class SetupProjectPerformanceMixin(object):
     @classmethod
     def make_mobile_worker(cls, username, domain=None):
         domain = domain or cls.domain
-        user = CommCareUser(username=username, domain=domain)
-        user.domain_membership = DomainMembership(domain=domain)
+        user = CommCareUser.create(domain, username, '123')
         doc = user._doc
         doc['username.exact'] = doc['username']
         UserESFake.save_doc(doc)
@@ -186,6 +185,7 @@ class ProjectHealthDashboardTest(SetupProjectPerformanceMixin, TestCase):
     def tearDownClass(cls):
         super(ProjectHealthDashboardTest, cls).tearDownClass()
         cls.class_teardown()
+        cls.web_user.delete()
         GroupESFake.reset_docs()
 
     @classmethod

